include/deal.II/lac/linear_operator.h

```diff
@@ -722,6 +723,72 @@ inverse_operator(const LinearOperator<Range, Domain, Payload> &op,
   return return_op;
 }
 
+
+/**
+ * @relatesalso LinearOperator
+ *
+ * Variant of above function that takes an rvalue-reference as
+ * preconditioner argument. This function takes care of instances where
+ * either a LinearOperator is created as a temporary argument, or where for
+ * example <code>PreconditionIdentity()</code> is supplied.
+ *
+ * In this case, instead of storing a reference to the preconditioner as is
+ * done in the previous function, we store a full copy of the
+ * preconditioner instead.
```

Comment:
This requires that these objects can be copied. That's not obvious to me in general. Have you tried a test where an object can not be copied?

Reply:
@bangerth Yes, I did: The old version of this PR would very clearly bail out in the corresponding copy constructor (for example for `sparse_ilu`), or the latest in the corresponding `vmult` because the object wouldn't have been initialized (for example for `precondition_jacobi`, etc).